herzog/__init__.py

```diff
@@ -38,10 +39,51 @@ def __exit__(self, *args, **kwargs):
                     del __main__.__dict__[key]
             __main__.__dict__.update(self._state_modules)
 
-def generate(handle: TextIO):
-    cells = [obj.to_ipynb_cell() for obj in parse_cells(handle)
+def load_ipynb_cells(ipynb: TextIO) -> List[Dict[Any, Any]]:
+    try:
+        cells = json.loads(ipynb.read()).get('cells', None)
+    except Exception:
+        print(f'Check that "{ipynb}" is a valid ipynb file.', file=sys.stderr)
+        raise
+
+    if not cells:
+        print(f'Python notebook: "{ipynb}" has no cells.\nNothing to be done.  Exiting... ')
+        exit()
+    return cells
+
+def generate(handle: TextIO, input_type: Optional[str] = 'herzog') -> str:
+    if input_type == 'herzog':
+        return translate_to_ipynb(handle)
+    elif input_type == 'ipynb':
+        return translate_to_herzog(handle)
+    else:
+        raise NotImplementedError(f'Input type: "{input_type}" not supported.')
+
+def translate_to_ipynb(herzog_handle: TextIO, indent: int = 2) -> str:
+    cells = [obj.to_ipynb_cell() for obj in parse_cells(herzog_handle)
              if obj.has_ipynb_representation]
     with open(os.path.join(os.path.dirname(__file__), "data", "python_3_boiler.json")) as fh:
         boiler = json.loads(fh.read())
-    ipynb = dict(cells=cells, **boiler)
+    ipynb = json.dumps(dict(cells=cells, **boiler), indent=indent)
     return ipynb
+
+def translate_to_herzog(ipynb_handle: TextIO, indent: int = 4) -> str:
+    cells = load_ipynb_cells(ipynb_handle)
+    python_prefix = ' ' * indent
+    markdown_prefix = python_prefix + '# '
```

Comment:
Herzog uses docstrings to encode markdown, not comments.

See the [readme example](https://github.com/xbrianh/herzog#example)

Reply:
I'm concerned that the current docstring implementation may not be deterministic.  I'll make a separate PR where I hope to make my points clear.

Reply:
Opened in #27.